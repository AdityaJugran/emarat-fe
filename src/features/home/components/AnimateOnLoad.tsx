import { TailwindClassesNumbers } from 'lib/types';
import { useState, useEffect } from 'react';

export type PropsType = {
	startFrom: 'top' | 'left' | 'right';
	strength?: TailwindClassesNumbers;
	children: React.ReactNode;
};

export default function AnimateOnLoad({
	startFrom,
	strength = 60,
	children,
}: PropsType) {
	const initialAnimationClasses = {
		top: `-translate-y-${strength}`,
		left: `-translate-x-${strength}`,
		right: `translate-x-${strength}`,
	};

	const [animationClasses, setAnimationClasses] = useState<string>(
		initialAnimationClasses[startFrom]
	);

	useEffect(() => {
		setAnimationClasses('');

		return () => {
			setAnimationClasses(initialAnimationClasses[startFrom]);
		};
	}, []);

	return (
		<div
			className={`transform transition-transform delay-100 ease-in duration-500 ${animationClasses}`}
		>
			{children}
		</div>
	);
}
