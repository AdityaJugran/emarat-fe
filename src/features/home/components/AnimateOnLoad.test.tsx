import AnimateOnLoad from 'features/home/components/AnimateOnLoad';
import renderer from 'react-test-renderer';

it('Animate On Load', () => {
	const tree = renderer
		.create(<AnimateOnLoad startFrom="left">mock child</AnimateOnLoad>)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
