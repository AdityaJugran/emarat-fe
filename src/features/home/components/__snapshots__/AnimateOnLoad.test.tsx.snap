// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animate On Load 1`] = `
<div
  className="transform transition-transform delay-100 ease-in duration-500 -translate-x-60"
>
  mock child
</div>
`;
